bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -2455,6 +2518,59 @@ public void setAutoExpandLevel(int level) {
 		expandToLevel = level;
 	}
 
+	/**
+	 * Sets the level of auto expand on paths of consecutive single children.
```

Comment:
I find this description hardly comprehensible since you rely on intuition for understanding what a "path of consecutive single children" is. What about something like: "Sets the maximum number of levels to automatically expand whenever a widget is expanded that only has a single child element. The expansion is recursively applied for the given number of levels if the widget at each of the according levels only has a single child."

Reply:
What information then still is missing is _when_ this mechanism is applied? It cannot be whenever a node is expanded, as the functionality would then apply recursively. It can also not be whenever a node is not expanded by this mechanism, as, e.g., the `expandToLevel` functionality expands nodes in a way that this new auto-expand functionality is not applied. The documentation should precisely state when the expansion happens. Right now, this is whenever an `SWT.EXPAND` event occurs on the underlying tree and whenever the `setExpandedState` method is called (which in my opinion is not correct, see according comment).